Source/Core/Common/ArmCPUDetect.cpp

```diff
@@ -9,24 +9,100 @@
 #include <string>
 #include <thread>
 
-#if !defined(_WIN32) && !defined(__APPLE__)
+#ifdef __APPLE__
+#include <sys/sysctl.h>
+#elif defined(_WIN32)
+#include <Windows.h>
+#include <intrin.h>
+#else
 #ifndef __FreeBSD__
 #include <asm/hwcap.h>
 #endif
 #include <sys/auxv.h>
-#include <unistd.h>
 #endif
 
 #include <fmt/format.h>
 
 #include "Common/CommonTypes.h"
 #include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
 
-#ifndef WIN32
+#ifdef __APPLE__
+
+static bool ReadProcessorString(std::string* value)
+{
+  const char* name = "machdep.cpu.brand_string";
+  size_t value_len = 0;
+  if (sysctlbyname(name, nullptr, &value_len, nullptr, 0) != ENOMEM)
```

Comment:
Tested this on an M1 Mac Mini. `0` will be returned if you specify `nullptr` as `oldp`, not `ENOMEM`.

Reply:
thanks for checking, should work now